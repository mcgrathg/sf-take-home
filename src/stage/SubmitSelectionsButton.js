import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import colors from '../config/colors';

const StyledButton = styled.button`
  display: inline-block;
  min-width: 13rem;
  height: 3.25rem;
  border: 1px solid ${colors.watermelon};
  padding: 0.5rem 1rem;
  color: ${colors.watermelon};
  background-color: $white;
  transition: all 0.25s ease-in;

  @media (min-width: 500px) {
    padding: 1rem 1.5rem;
  }

  &:hover,
  &:focus {
    color: ${colors.white};
    background-color: ${colors.watermelon};
  }

  &:active {
    transform: scale(0.98);
    transition: none;
  }

  &:focus {
    box-shadow: 0 0 0 3px ${colors.greyLight}, 0 0 0 5px ${colors.watermelon};
  }
`;

const SubmitSelectionsButton = ({ onClickHandler, text }) => (
  <StyledButton type="button" onClick={onClickHandler}>
    {text}
  </StyledButton>
);

SubmitSelectionsButton.propTypes = {
  onClickHandler: PropTypes.func.isRequired,
  text: PropTypes.string,
};

SubmitSelectionsButton.defaultProps = {
  text: 'â†’',
};

export default SubmitSelectionsButton;
