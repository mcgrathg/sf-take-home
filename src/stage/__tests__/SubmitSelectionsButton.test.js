import React from 'react';
import { render } from 'react-testing-library';
import { noOp } from 'test-utils';
import SubmitSelectionsButton from '../SubmitSelectionsButton';

describe('SubmitSelectionsButton', () => {
  it('renders', () => {
    const { container } = render(
      <SubmitSelectionsButton onClickHandler={noOp} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
