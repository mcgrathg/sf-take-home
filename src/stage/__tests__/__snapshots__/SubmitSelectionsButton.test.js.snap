// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitSelectionsButton renders 1`] = `
.c0 {
  width: 13rem;
  max-width: 100%;
  height: 3.25rem;
  border: 1px solid #d43f43;
  padding: 0.5rem 1rem;
  color: #d43f43;
  background-color: #fff;
  -webkit-transition: all 0.25s ease-in;
  transition: all 0.25s ease-in;
}

.c0:hover,
.c0:focus {
  color: #fff;
  background-color: #d43f43;
}

.c0:active {
  -webkit-transform: scale(0.98);
  -ms-transform: scale(0.98);
  transform: scale(0.98);
  -webkit-transition: none;
  transition: none;
}

.c0:focus {
  box-shadow: 0 0 0 3px #f2f0ee,0 0 0 5px #d43f43;
}

<button
  class="c0"
  title="Next"
  type="button"
>
  â†’
</button>
`;
